backend/routes/UserRoutes.js

```diff
@@ -54,7 +54,7 @@ router.get("/:id", (req, res, next) => {
 
   UserModel.findById(
     req.params.id,
```

Comment:
This is the main change to make if we decide to switch away from storing the user's id in localStorage:

```suggestion
    req.params.id ?? req.user._id,
```